minimesos/src/main/java/com/containersol/minimesos/junit/MesosClusterTestRule.java

```diff
@@ -1,22 +1,34 @@
 package com.containersol.minimesos.junit;
 
+import com.containersol.minimesos.MinimesosException;
 import com.containersol.minimesos.cluster.MesosCluster;
 import com.containersol.minimesos.cluster.MesosClusterFactory;
-import com.containersol.minimesos.main.factory.MesosClusterContainersFactory;
-import com.containersol.minimesos.mesos.ClusterArchitecture;
+import com.containersol.minimesos.mesos.MesosClusterContainersFactory;
 import org.junit.rules.TestRule;
 import org.junit.runner.Description;
 import org.junit.runners.model.Statement;
 
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+
 /**
  * JUnit Rule extension of Mesos Cluster to use in JUnit.
  *
  * TODO: see https://github.com/ContainerSolutions/minimesos/issues/8 for completion
  */
-public class MesosClusterTestRule extends MesosCluster implements TestRule {
+public class MesosClusterTestRule implements TestRule {
+
+    private MesosClusterFactory factory = new MesosClusterContainersFactory();
+
+    private final MesosCluster cluster;
 
-    public MesosClusterTestRule(ClusterArchitecture clusterArchitecture) {
-        super(clusterArchitecture.getClusterConfig(), clusterArchitecture.getClusterContainers().getContainers());
+    public MesosClusterTestRule(File minimesosFile) {
```

Comment:
It would be great if you can make the default constructor private and change this one into
`public static MesosClusterTestRule fromFileI(String minimesosFilePath)`